### PR TITLE
Fix #123

### LINKED ISSUE #123: Doesn't support GOPATH list
Hi! I'm giving a test ride to Kallax and I hit my first minor bump in the road. It's not critical as there is an easy workaround (`export GOPATH=/my/first/go/path`), but it is an inconvenience.

As described [in the Go documentation](https://golang.org/cmd/go/#hdr-GOPATH_environment_variable), GOPATH can be a list (`:`-seperated on UNIX).

But currently the GOPATH is treated as a single value ([here](https://github.com/src-d/go-kallax/blob/master/generator/processor.go#L462) and [here](https://github.com/src-d/go-kallax/blob/master/generator/template.go#L433)).

In my env it generates errors like this : 
```bash
> echo $GOPATH
/my/first/go/path:/my/second/go/path
> go generate .
panic: open /my/first/go/path:/my/second/go/path/src/gopkg.in/src-d/go-kallax.v1/generator/templates/base.tgo: no such file or directory

goroutine 1 [running]:
gopkg.in/src-d/go-kallax.v1/generator.loadTemplateText(0x727b86, 0x12, 0x73a801, 0xc420049ed0)
        /my/first/go/path/src/gopkg.in/src-d/go-kallax.v1/generator/template.go:436 +0xe0
gopkg.in/src-d/go-kallax.v1/generator.makeTemplate(0x711a4e, 0x4, 0x727b86, 0x12, 0x1)
        /my/first/go/path/src/gopkg.in/src-d/go-kallax.v1/generator/template.go:448 +0x39
gopkg.in/src-d/go-kallax.v1/generator.init()
        /my/first/go/path/src/gopkg.in/src-d/go-kallax.v1/generator/template.go:457 +0x126
main.init()
        /my/first/go/path/src/gopkg.in/src-d/go-kallax.v1/generator/cli/kallax/cmd.go:80 +0x58
kallax.go:14: running "kallax": exit status 2
```

### PR DESCRIPTION
**Pull request by hectorj**
To test it, `export GOPATH=~/go:/tmp/whatever` and `go test -v ./generator/`. On master it fails : 

```
→ go test -v ./generator/
# gopkg.in/src-d/go-kallax.v1/generator
generator/processor_test.go:12:2: cannot find package "srcd.works/go-parse-utils.v1" in any of:
	/home/hectorj/.local/share/umake/go/go-lang/src/srcd.works/go-parse-utils.v1 (from $GOROOT)
	/home/hectorj/go/src/srcd.works/go-parse-utils.v1 (from $GOPATH)
	/tmp/whatever/src/srcd.works/go-parse-utils.v1
FAIL	gopkg.in/src-d/go-kallax.v1/generator [setup failed]
```

After this PR it passes

### CODE REVIEW

**Comment by erizocosmico** [comment]
Thanks for the contribution! We're at a conference right now, but will try to take a look at it as soon as possible.
On Fri, 21 Apr 2017 at 08:31, codecov[bot] <notifications@github.com<mailto:notifications@github.com>> wrote:
Codecov<https://codecov.io/gh/src-d/go-kallax/pull/130?src=pr&el=h1> Report

Merging #130<https://codecov.io/gh/src-d/go-kallax/pull/130?src=pr&el=desc> into master<https://codecov.io/gh/src-d/go-kallax/commit/dce9f7c285e7f341474e16f53098917104013474?src=pr&el=desc> will decrease coverage by 0.13%.
The diff coverage is 69.23%.

[Impacted file tree graph]<https://codecov.io/gh/src-d/go-kallax/pull/130?src=pr&el=tree>

@@            Coverage Diff             @@
##           master     #130      +/-   ##
==========================================
- Coverage   80.35%   80.22%   -0.14%
==========================================
  Files          15       15
  Lines        2785     2797      +12
==========================================
+ Hits         2238     2244       +6
- Misses        375      378       +3
- Partials      172      175       +3

Impacted Files<https://codecov.io/gh/src-d/go-kallax/pull/130?src=pr&el=tree>   Coverage Δ
generator/template.go<https://codecov.io/gh/src-d/go-kallax/pull/130?src=pr&el=tree#diff-Z2VuZXJhdG9yL3RlbXBsYXRlLmdv>  92.37% <100%> (-0.53%)  ⬇️
generator/processor.go<https://codecov.io/gh/src-d/go-kallax/pull/130?src=pr&el=tree#diff-Z2VuZXJhdG9yL3Byb2Nlc3Nvci5nbw==>     87.96% <63.63%> (-1.1%) ⬇️
________________________________

Continue to review full report at Codecov<https://codecov.io/gh/src-d/go-kallax/pull/130?src=pr&el=continue>.

Legend - Click here to learn more<https://docs.codecov.io/docs/codecov-delta>
Δ = absolute <relative> (impact), ø = not affected, ? = missing data
Powered by Codecov<https://codecov.io/gh/src-d/go-kallax/pull/130?src=pr&el=footer>. Last update dce9f7c...5452456<https://codecov.io/gh/src-d/go-kallax/pull/130?src=pr&el=lastupdated>. Read the comment docs<https://docs.codecov.io/docs/pull-request-comments>.

File: generator/processor.go [hunk 1]
```diff
@@ -459,14 +459,29 @@ func joinDirectory(directory string, files []string) []string {
...
+	return pkg.Dir, nil
+}
 
 func typeName(typ types.Type) string {
 	return removeGoPath(typ.String())
 }
 
 func removeGoPath(path string) string {
-	return strings.Replace(path, goPath+"/src/", "", -1)
+	for _, goPath := range goPaths {
```
Comment (hectorj) [inline generator/processor.go]: Might be better to use a [strings.Replacer](https://golang.org/pkg/strings/#Replacer) (though [strings.NewReplacer](https://golang.org/pkg/strings/#NewReplacer) is not very practical to use in this case)

File: generator/processor.go [hunk 2]
```diff
@@ -459,14 +459,29 @@ func joinDirectory(directory string, files []string) []string {
 	return result
 }
 
-var goPath = os.Getenv("GOPATH")
+var goPaths = filepath.SplitList(build.Default.GOPATH)
+
+func getPkgSrcDir(importPath string) (string, error) {
+	wd, err := os.Getwd()
```
Comment (hectorj) [inline generator/processor.go]: I'm not really sure that the current working directory is the good thing to pass to build.Import, but it seems to do the job

**Comment by hectorj** [comment]
I merged the test change from the `gopath-fix` branch, and now the travis build is failing :thinking: 

But I can't reproduce that fail on my local...

**Comment by erizocosmico** [comment]
This approach only works while using `go generate` because the working directory is the one of the package being generated. But if you run `kallax gen` by hand this does not work.

**Comment by erizocosmico** [comment]
Closing this in favor of #138

**Comment by hectorj** [comment]
ah, I forgot I had https://github.com/src-d/go-parse-utils/pull/6 on my local, might have been what made my tests pass.

Anyway, #138 looks good :+1:

**Pull request closed without merging**